### PR TITLE
fix: Fixed Compute-Image dependency - `avm/res/compute/image`

### LINKED ISSUE #1625: [Failed pipeline] avm.res.compute.image
Failed run: https://github.com/Azure/bicep-registry-modules/actions/runs/8596232887

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- The current combination of deployment script PowerShell version `8.0` and the installed [Az.ImageBuilder](https://www.powershellgallery.com/packages/Az.ImageBuilder/0.4.1) version `0.4.1` were incompatible (as the latter needs at least Az.Accounts version `2.19.0` installed, which is not available in pwsh `8.0`)
- Upgrading just the Deployment Script's PowerShell version does not work, as the [latest available](https://[mcr.microsoft.com/v2/azuredeploymentscripts-powershell/tags/list](https://mcr.microsoft.com/v2/azuredeploymentscripts-powershell/tags/list)) is `11.5` which only has Az.Accounts version `2.13.0` installed
- So, to solve the issue I updated the implementation to install Az.ImageBuilder version `0.4.0`  explicitely which has only a dependency on Az.Accounts version `2.13.0`.
- In the same way, I updated the installation of the `Az.Storage` module (which in its latest version required Az.Accounts `3.0.0`)

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.compute.image](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.compute.image.yml)      |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

**Comment by AlexanderSehr** [comment]
Just to 7 retries 😄

File: avm/utilities/pipelines/e2eValidation/resourceDeployment/New-TemplateDeployment.ps1 [hunk 1]
```diff
@@ -428,7 +428,7 @@ function New-TemplateDeployment {
         [switch] $DoNotThrow,
 
         [Parameter(Mandatory = $false)]
-        [int] $RetryLimit = 3,
+        [int] $RetryLimit = 1,
```
[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceDeployment/New-TemplateDeployment.ps1:431]: No. It was just for testing. Pardon

[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceDeployment/New-TemplateDeployment.ps1]: undone @eriqua

**Comment by eriqua** [comment]
@tony-box @AlexanderSehr I'd usually leave the approval to the module owner, but in this case the core team is required since e2e shared utilities are updated. @tony-box anything against that fix? Otherwise I'll approve

**Comment by AlexanderSehr** [comment]
Then please somebody approve 😄

**Review by tony-box — APPROVED** [review]
oops, didn't know I could review this one. Looks good!

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**